src/Microsoft.Framework.DependencyInjection.Abstractions/ServiceDescriptor.cs

```diff
@@ -72,6 +72,24 @@ private ServiceDescriptor(Type serviceType, ServiceLifetime lifetime)
         /// <inheritdoc />
         public Func<IServiceProvider, object> ImplementationFactory { get; }
 
+        public Type GetImplementationType()
```

Comment:
The name for ```GetImplementationType()``` is kind of awful but ```ImplementaitonType``` is already taken with different semantics.

Might also be better as an auto property set on construction